internal/pkg/cache/add.go

```diff
@@ -45,13 +46,28 @@ func (c *Cache) Add(opts *AddOpts) (cachedRegistry *Registry, err error) {
 	// to look up the source URL, but invalid packs metadata can mess that up.
 	// Throw error if registry source is not defined
 	if opts.Source == "" {
-		err = errors.ErrRegistrySourceRequired
-		return
+		return cachedRegistry, errors.ErrRegistrySourceRequired
 	}
 
-	cachedRegistry, err = c.addFromURI(opts)
+	cachedRegistry, err := c.addFromURI(opts)
+	if err != nil {
+		return cachedRegistry, err
+	}
 
-	return
+	if cachedRegistry != nil {
+		// Store a metadata JSON file for the cached registry
+		b, err := json.MarshalIndent(cachedRegistry, "", "  ")
+		if err != nil {
+			return cachedRegistry, err
+		}
+
+		metaPath := fmt.Sprintf("%s/%s/metadata.json", c.cfg.Path, opts.RegistryName)
```

Comment:
```suggestion
		metaPath := fmt.Sprintf("%s/%s/metadata.json", c.cfg.Path, opts.RegistryName)
```

use `filepath.Join` for safety and cross-platform